### PR TITLE
Edit formatting of name and address parameter

### PR DESCRIPTION
**Pull request by jktang14**
Format for name and address parameters have now been change to allow hyphens and spaces between characters. Only 1 hyphen and space should be between each character.

Hyphens cannot appear at the start and end of names.

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Phone.java [hunk 1]
```diff
@@ -30,7 +31,8 @@ public Phone(String phone) {
      * Returns true if a given string is a valid phone number.
      */
     public static boolean isValidPhone(String test) {
-        return test.matches(VALIDATION_REGEX);
+        String trimmedTest = test.trim();
```
Comment (Dinoman44) [inline src/main/java/seedu/address/model/person/Phone.java:34]: I think the point of doing the exact math was to make sure there is no leading/trailing whitespace right?

**Review by Dinoman44 — APPROVED** [review]

File: src/test/java/seedu/address/model/person/NameTest.java [hunk 2]
```diff
@@ -29,13 +29,21 @@ public void isValidName() {
         assertFalse(Name.isValidName(" ")); // spaces only
         assertFalse(Name.isValidName("^")); // only non-alphanumeric characters
         assertFalse(Name.isValidName("peter*")); // contains non-alphanumeric characters
+        assertFalse(Name.isValidName("David-")); // Hyphen at end of name
```
Comment (rahula1008) [inline src/test/java/seedu/address/model/person/NameTest.java:32]: I like the thorough use of test cases here

**Review by rahula1008 — APPROVED** [review]

File: src/test/java/seedu/address/model/person/PhoneTest.java [hunk 3]
```diff
@@ -30,12 +30,21 @@ public void isValidPhone() {
         assertFalse(Phone.isValidPhone("91")); // less than 3 numbers
         assertFalse(Phone.isValidPhone("phone")); // non-numeric
         assertFalse(Phone.isValidPhone("9011p041")); // alphabets within digits
-        assertFalse(Phone.isValidPhone("9312 1534")); // spaces within digits
+        assertFalse(Phone.isValidPhone("-934")); // Hyphen at the start
+        assertFalse(Phone.isValidPhone("934-")); // Hyphen at the end
+        assertFalse(Phone.isValidPhone("934--999")); // Double hyphen between digits
+        assertFalse(Phone.isValidPhone("934  999")); // Double space between digits
+        assertFalse(Name.isValidName("999- 934")); // Hyphen and space between 2 characters
```
Comment (rahula1008) [inline src/test/java/seedu/address/model/person/PhoneTest.java:37]: Like the thorough use of test cases here

File: src/main/java/seedu/address/model/person/Name.java [hunk 4]
```diff
@@ -10,13 +10,14 @@
...
-            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+            "Names should only contain alphanumeric characters, spaces and hyphens.\n"
+                    + "Only 1 space or hyphen is allowed between each alphanumeric character. It should not be blank.";
 
     /*
      * The first character of the address must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
      */
-    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}]+([ -][\\p{Alnum}]+)*\\s*";
```
Comment (rahula1008) [inline src/main/java/seedu/address/model/person/Name.java:20]: LGTM

File: src/test/java/seedu/address/model/person/PersonTest.java [hunk 5]
```diff
@@ -45,10 +45,10 @@ public void isSamePerson() {
         Person editedBob = new PersonBuilder(BOB).withName(VALID_NAME_BOB.toLowerCase()).build();
         assertTrue(BOB.isSamePerson(editedBob));
 
-        // name has trailing spaces, all other attributes same -> returns false
+        // name has trailing spaces, all other attributes same -> returns True
```
Comment (rahula1008) [inline src/test/java/seedu/address/model/person/PersonTest.java:48]: LGTM

**Review by rahula1008 — APPROVED** [review]

**Pull request merged**